ev3dev2/led.py

```diff
@@ -346,3 +385,231 @@ def all_off(self):
 
         for led in self.leds.values():
             led.brightness = 0
+
+    def reset(self):
+        """
+        Put all LEDs back to their default color
+        """
+
+        if not self.leds:
+            return
+
+        for group in self.led_groups:
+            self.set_color(group, LED_DEFAULT_COLOR)
+
+    def animate_stop(self):
+        """
+        Signal the current animation thread to exit and wait for it to exit
+        """
+
+        if self.animate_thread_id:
+            self.animate_thread_stop = True
+
+            while self.animate_thread_id:
+                pass
+
+    def animate_police_lights(self, color1, color2, group1='LEFT', group2='RIGHT', sleeptime=0.5, duration=5, block=True):
+        """
+        Cycle the ``group1`` and ``group2`` LEDs between ``color1`` and ``color2``
+        to give the effect of police lights.  Alternate the ``group1`` and ``group2``
+        LEDs every ``sleeptime`` seconds.
+
+        Animate for ``duration`` seconds.  If ``duration`` is None animate for forever.
+
+        Example:
+
+        .. code-block:: python
+
+            from ev3dev2.led import Leds
+            leds = Leds()
+            leds.animate_police_lights('RED', 'GREEN')
+        """
+
+        def _animate_police_lights():
+            self.all_off()
+            even = True
+            start_time = dt.datetime.now()
+
+            while True:
+                if even:
+                    self.set_color(group1, color1)
+                    self.set_color(group2, color2)
+                else:
+                    self.set_color(group1, color2)
+                    self.set_color(group2, color1)
+
+                if self.animate_thread_stop or duration_expired(start_time, duration):
+                    break
+
+                even = not even
+                sleep(sleeptime)
+
+            self.animate_thread_stop = False
+            self.animate_thread_id = None
+
+        self.animate_stop()
+
+        if block:
+            _animate_police_lights()
+        else:
+            self.animate_thread_id = _thread.start_new_thread(_animate_police_lights, ())
+
+    def animate_flash(self, color, groups=('LEFT', 'RIGHT'), sleeptime=0.5, duration=5, block=True):
```

Comment:
Can this be implemented in terms of `animate_cycle`? (i.e., one of the colors is the provided one and the other is `BLACK`)

Reply:
I don't think it can.  ``animate_cycle`` is changing all of the LEDs from one color to the next through a list of colors where ``animate_police_lights`` is toggling two groups of LEDs back and forth between two colors.

Reply:
This was about `animate_flash` rather than police lights. Not a big issue though.